pkg/instance/repository.go

```diff
@@ -21,6 +21,37 @@ type repository struct {
 	instanceParameterEncryptionKey string
 }
 
+func (r repository) SaveDeployment(deployment *model.Deployment) error {
+	// TODO: Do we need the option to save nested entities... Yes, if we create the deployment from a preset we need to store all the links as well
+	//err := r.db.Session(&gorm.Session{FullSaveAssociations: true}).Save(deployment).Error
+	err := r.db.Create(&deployment).Error
+	if err != nil {
+		if errors.Is(err, gorm.ErrDuplicatedKey) {
+			return errdef.NewDuplicated("a deployment named %q already exists", deployment.Name)
+		}
+		return fmt.Errorf("failed to save deployment: %v", err)
```

Comment:
the comment here also applied to all other repository methods 😅

Reply:
What comment?